substrate/frame/treasury/src/lib.rs

```diff
@@ -987,6 +989,19 @@ impl<T: Config<I>, I: 'static> OnUnbalanced<NegativeImbalanceOf<T, I>> for Palle
 	}
 }
 
+/// Implement the `OnUnbalanced` handler for [`frame_support::traits::fungible`] trait currency.
```

Comment:
Probably unrelated to the PR?

Reply:
Staking uses fungible now, and treasury is the slash handler for it. While treasury still uses Currency which is not compatible with fungibles. `FungibleCompat` adds a way for treasury to manage fungible imbalance.